fcrepo-persistence-ocfl/src/main/java/org/fcrepo/persistence/ocfl/OCFLPersistentStorageSession.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * Licensed to DuraSpace under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional information
+ * regarding copyright ownership.
+ *
+ * DuraSpace licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except in
+ * compliance with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.fcrepo.persistence.ocfl;
+
+import java.time.Instant;
+
+import org.fcrepo.kernel.api.exception.CannotCreateResourceException;
+import org.fcrepo.kernel.api.exception.ItemNotFoundException;
+import org.fcrepo.kernel.api.exception.RepositoryRuntimeException;
+import org.fcrepo.kernel.api.models.FedoraResource;
+import org.fcrepo.kernel.api.models.Tombstone;
+import org.fcrepo.persistence.api.PersistentStorageSession;
+import org.fcrepo.persistence.api.exceptions.PersistentStorageException;
+
+/**
+ * OCFL Persistent Storage class.
+ *
+ * @author whikloj
+ * @since 2019-09-20
+ */
+public class OCFLPersistentStorageSession implements PersistentStorageSession {
+
+    /**
+     * Fedora transaction id for the session.
+     */
+    private final String sessionId;
+
+    /**
+     * Constructor
+     *
+     * @param txId Fedora transaction id.
+     */
+    protected OCFLPersistentStorageSession(final String txId) {
+        this.sessionId = txId;
+    }
+
+    /**
+     * Constructor
+     */
+    protected OCFLPersistentStorageSession() {
+        this.sessionId = null;
+    }
+
+    @Override
+    public String getTxId() {
+        return this.sessionId;
+    }
+
+    @Override
+    public FedoraResource create(final FedoraResource resource) throws CannotCreateResourceException {
+        actionNeedsWrite();
+        return null;
+    }
+
+    @Override
+    public FedoraResource update(final FedoraResource resource) throws CannotCreateResourceException {
+        actionNeedsWrite();
+        return null;
+    }
+
+    @Override
+    public Tombstone delete(final FedoraResource resource) throws RepositoryRuntimeException {
+        actionNeedsWrite();
+        return null;
+    }
+
+    @Override
+    public FedoraResource read(final String identifier) throws ItemNotFoundException {
+        actionNeedsWrite();
+        return null;
+    }
+
+    @Override
+    public FedoraResource read(final String identifier, final Instant version) throws ItemNotFoundException {
+        if (!isReadOnly()) {
+            throw new PersistentStorageException("Can't read versions in a session");
```

Comment:
It seems to me that you should be allowed to access this method within a session.  It will simply return a result that is identical to the same invocation on a read-only session.